tests/robustness/coverage/coverage_test.go

```diff
@@ -16,20 +16,111 @@ package coverage_test
 
 import (
 	"bytes"
+	"cmp"
 	"encoding/json"
 	"fmt"
+	"maps"
 	"os"
 	"path/filepath"
+	"slices"
+	"strconv"
 	"strings"
 	"testing"
 
+	gocmp "github.com/google/go-cmp/cmp"
 	"github.com/olekukonko/tablewriter"
 	"github.com/olekukonko/tablewriter/tw"
 	traceservice "go.opentelemetry.io/proto/otlp/collector/trace/v1"
 	tracev1 "go.opentelemetry.io/proto/otlp/trace/v1"
 	"google.golang.org/protobuf/encoding/protojson"
 )
 
+type refOp struct {
+	// args splits traces into buckets based on Matched results.
+	args map[string]Matched
+	// wantArgs lists the expected buckets stored in args.
+	wantArgs map[string]struct{}
+	// methods tries to associate Matched with method name.
+	methods map[string]Matched
+	// knownBypasses filters out traces from the args analysis.
+	knownBypasses map[string]Matched
+}
+
+var referenceUsageOfEtcdAPI = map[string]refOp{
+	"etcdserverpb.KV/Range": {
+		// All calls should go through etcd-k8s interface
+		args: map[string]Matched{
+			"countOnly": isCountOnly,
+			"keysOnly":  isKeysOnly,
+			"limit":     isLimitSet,
+			"limit=1":   keyIsEqualInt("limit", 1),
+			"rangeEnd":  isRangeEndSet,
+			"rev":       isRevisionSet,
+		},
+		wantArgs: map[string]struct{}{
+			"countOnly=false,keysOnly=false,limit=false,limit=1=false,rangeEnd=true,rev=false": {},
+			"countOnly=false,keysOnly=false,limit=true,limit=1=false,rangeEnd=true,rev=false":  {},
+			"countOnly=false,keysOnly=false,limit=true,limit=1=false,rangeEnd=true,rev=true":   {},
+			"countOnly=false,keysOnly=false,limit=true,limit=1=true,rangeEnd=false,rev=false":  {},
+			"countOnly=false,keysOnly=false,limit=true,limit=1=true,rangeEnd=true,rev=true":    {},
+			"countOnly=true,keysOnly=false,limit=false,limit=1=false,rangeEnd=true,rev=false":  {},
+		},
+		methods: map[string]Matched{
+			"Get": andMatcher(
+				keyIsEqualInt("limit", 1),
+				notMatcher(isRangeEndSet),
+			),
+			"Count": andMatcher(
+				isCountOnly,
+				isRangeEndSet,
+				notMatcher(orMatcher(isKeysOnly, isLimitSet, isRevisionSet)),
+			),
+			"List": andMatcher(
+				isRangeEndSet,
+				notMatcher(orMatcher(isKeysOnly, isCountOnly, keyIsEqualInt("limit", 1))),
+			),
+			"List(ConsistentRead)": andMatcher(
+				keyIsEqualInt("limit", 1),
+				isRangeEndSet,
+				isRevisionSet,
+			),
+		},
+		knownBypasses: map[string]Matched{
```

Comment:
I you modeled key pattern you would not need separate `knownBypasses`.

Reply:
Large table is not a problem as long at it is readable. Don't think `shashes` is readable.
I would suggest to have `pattern` with values like:
* `/healthz` - Health
* `/registry/{resource}` - ListAll
* `/registry/{resource}/{namespace}` ListNamespace
* `/registry/{resource}/{namespace}/{name}` - NonRecursiveList
* `/registry/{resource}/{namespace}/{name}` - Get

Reply:
Maybe because key pattern as the most important column should be first?

Reply:
Updated the main comment with new test outputs